internal/constants/scripts.go

```diff
@@ -179,6 +177,14 @@ function copy_nmconnection_files_to_nm_config_dir() {
     fi
   done
 
+  # If host_dir is not empty then delete the default nmconnection
+  if [ "$(ls -A ${host_dir})" ]; then
```

Comment:
find_host_directory_by_mac_address returns an error if it can't find a directory, so I think you can just delete the default right after that function.